src/Analyzer.Core/TemplateAnalyzer.cs

```diff
@@ -114,42 +166,82 @@ public IEnumerable<IEvaluation> AnalyzeTemplate(string template, string template
                 ResourceMappings = armTemplateProcessor.ResourceMappings,
                 TemplateIdentifier = templateFilePath,
-                IsBicep = isBicep,
-                SourceMap = sourceMap
+                IsBicep = parentContext.IsBicep,
+                SourceMap = parentContext.SourceMap,
+                PathPrefix = pathPrefix,
+                ParentContext = parentContext
             };
 
             try
             {
                 IEnumerable<IEvaluation> evaluations = this.jsonRuleEngine.AnalyzeTemplate(templateContext);
                 evaluations = evaluations.Concat(this.powerShellRuleEngine.AnalyzeTemplate(templateContext));
 
-                // For each rule we don't want to report the same line more than once
-                // This is a temporal fix
-                var evalsToValidate = new List<IEvaluation>();
-                var evalsToNotValidate = new List<IEvaluation>();
-                foreach (var eval in evaluations)
+                // Recursively handle nested templates 
+                var jsonTemplate = JObject.Parse(populatedTemplate);
+                var processedTemplateResources = templatejObject.InsensitiveToken("resources");
+
+
+                for (int i = 0; i < processedTemplateResources.Count(); i++)
                 {
-                    if (!eval.Passed && eval.Result != null)
-                    {
-                        evalsToValidate.Add(eval);
-                    }
-                    else
+                    var currentProcessedResource = processedTemplateResources[i];
+
+                    if (currentProcessedResource.InsensitiveToken("type")?.ToString().Equals("Microsoft.Resources/deployments", StringComparison.OrdinalIgnoreCase) ?? false)
                     {
-                        evalsToNotValidate.Add(eval);
+                        var nestedTemplate = currentProcessedResource.InsensitiveToken("properties.template");
+                        if (nestedTemplate == null)
+                        {
+                            continue; // This is a Linked template 
+                        }
+                        var populatedNestedTemplate = nestedTemplate.DeepClone();
+
+                        // Check whether scope is set to inner or outer
+                        var scope = currentProcessedResource.InsensitiveToken("properties.expressionEvaluationOptions")?.InsensitiveToken("scope")?.ToString();
+                        string nextPathPrefix = nestedTemplate.Path;
+                                               
+                        if (scope == null || scope.Equals("outer", StringComparison.OrdinalIgnoreCase))
+                        {
+                            // Variables, parameters and functions inherited from parent template
+                            string functionsKey = populatedNestedTemplate.InsensitiveToken("functions")?.Parent.Path;
+                            functionsKey ??= "functions";
```

Comment:
This can be combined.
``` suggestion
                            string functionsKey = populatedNestedTemplate.InsensitiveToken("functions")?.Parent.Path ?? "functions";
```